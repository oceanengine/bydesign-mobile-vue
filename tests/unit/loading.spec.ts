import { mount } from '../index';
import { Loading as NativeLoading } from '@src';
const Loading = NativeLoading as any;
describe('Loading', () => {
  test('size prop', async () => {
    const wrapper = mount(Loading, {
      propsData: {
        size: 20,
      },
    });

    expect(wrapper).toMatchSnapshot();
  });
  test('default slot', () => {
    const wrapper = mount(Loading, {
      scopedSlots: {
        default: () => 'Text',
      },
    });

    expect(wrapper).toMatchSnapshot();
  });
  test('text prop', () => {
    const wrapper = mount(Loading, {
      propsData: {
        text: 'Text',
      },
    });

    expect(wrapper).toMatchSnapshot();
  });
});
