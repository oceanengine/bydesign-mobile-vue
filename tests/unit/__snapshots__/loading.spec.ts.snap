// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading default slot 1`] = `
<div class="byted-loading"><i class="byted-loading__icon byted-loading-circle">test-file-stub</i>
  <div class="byted-loading__text">Text</div>
</div>
`;

exports[`Loading size prop 1`] = `<div class="byted-loading"><i class="byted-loading__icon byted-loading-circle" style="width: 20px; height: 20px;">test-file-stub</i></div>`;

exports[`Loading text prop 1`] = `
<div class="byted-loading"><i class="byted-loading__icon byted-loading-circle">test-file-stub</i>
  <div class="byted-loading__text">Text</div>
</div>
`;
